### PR TITLE
[d3-shape] Update types to v2.0

### PR DESCRIPTION
**Pull request by Methuselah96**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/d3/d3-shape/releases/tag/v2.0.0
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

Related: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/38939

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## recharts ([<kbd>unpkg</kbd>](https://unpkg.com/browse/recharts@latest/))
was missing the following properties:
1. Layer
1. Sankey
1. Funnel
1. FunnelChart
1. Trapezoid

<details>
<summary>as well as these 5 other properties...</summary>
<p>Layer, Sankey, Funnel, FunnelChart, Trapezoid</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against a59549a524d4ae0571e0944998b5ce616182e984
</p>

File: types/d3-shape/index.d.ts [hunk 1]
```diff
@@ -2395,24 +2393,34 @@ export interface Symbol<This, Datum> {
...
+ * The second generic corresponds to the data type of the datum underlying the symbol.
+ *
+ * @param type The specified type.
+ * @param size The specified size.
  */
-export function symbol<Datum>(): Symbol<any, Datum>; // tslint:disable-line ban-types
+export function symbol<Datum = any>(
+    type?: SymbolType | ((this: any, d: Datum, ...args: any[]) => SymbolType),
+    size?: number | ((this: any, d: Datum, ...args: any[]) => number)
+): Symbol<any, Datum>; // tslint:disable-line ban-types
```
Comment (amcasey) [inline types/d3-shape/index.d.ts:2408]: Why was a suppression required here?

**Review by amcasey — CHANGES_REQUESTED** [review]

File: types/d3-shape/index.d.ts (hunk 1 above)
Comment (Methuselah96) [reply types/d3-shape/index.d.ts:2408]: I didn't add this code. I'll check to see if it can be removed.

Comment (Methuselah96) [reply types/d3-shape/index.d.ts]: It wasn't necessary. I've removed it.

**Review by amcasey — APPROVED** [review]
Thanks!

**Comment by Methuselah96** [comment]
Ready to merge

**Pull request merged**